### ISSUE TITLE
[Feature Request] Hitsound on input

### ISSUE THREAD

**Issue by Arcadris** [PizzaLovers007/AdofaiTweaks#19]
Currently the hitsounds for maps play perfectly on beat even if your input is off. I currently slightly emulate this by having a program that plays a sound on keypress and using this mod to turn off the normal hitsounds, but having the hitsounds set to play instead of a generic sound would be better.

**Comment by CrackThrough** [comment]
There are some nitpicks. That would require canceling all of the queued hitsounds. Also at some point level requires changing a hitsound to another, so it would be very difficult to implement it on its own.

This is all solvable by doing some big refactoring but the main problem is the sound itself is not synced. The game has a hard-bounded offset per each hitsounds, and some hitsounds like oneshot would no longer have the same sound if they were chopped off until the offset. If you look at how BMS participants did it, they had to chop the sounds on their own and provide that "early" sound on another hidden SFX note or last note and manually add delays, or the sounds would feel desynced.

This is a problem that rhythm games with dynamic hitsounds had to deal with, and is still unsolved. It is still "clear enough to code" but it just takes too much effort. I'm not sure about other developers, but I don't think we can do that anytime soon.

**Comment by CrackThrough** [comment]
Yep this is straight up impossible as of now. I will close this issue.
